### ISSUE TITLE
Move to Cython3 or activate binding=True for pyOpenMS

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#4921]
To have better introspection.

https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html

**Comment by jpfeuffer** [comment]
Add here:
https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/create_cpp_extension.py#L176

Why is it still language_level = 2 again? @hroest

**Comment by jpfeuffer** [comment]
I don't think building with language_level = 2 is a long-term solution. I assume we won't be able to build with newer cython versions and won't get any potential bugfixes etc.

**Comment by jpfeuffer** [comment]
This was the reason #4229

**Comment by jpfeuffer** [comment]
Also maybe `language_level=3str` helps with our string problems? idk

**Comment by jpfeuffer** [comment]
Ok. We cannot use binding=True since autowrap cannot wrap static methods correctly.

**Comment by jpfeuffer** [comment]
https://github.com/uweschmitt/autowrap/issues/35

**Comment by jpfeuffer** [comment]
However, now I think it looked worse than it is because I am using PyCharm. PyCharm creates fake python sources for extension modules to create function signatures that can be used for autocompletion. It tries to create a signature based on the docstring (which is generated nicely by autowrap). While this works well for simple methods (e.g. no overloaded methods) with simple param objects (e.g. no lists), it fails for anything more complex.

I am kind of happy with the docstrings for now, I just though something was broken with them. With autowrap we probably never will need "embed_signatures" (I wonder what the difference would be).

I am still wondering what binding=True would lead to. It should be able to get real signatures with that if we make autowrap decorate static methods. Seems to build a bit longer though and I read somewhere that there might be slight performance decreases.

**Comment by timosachsenberg** [comment]
I think we moved to cython 3 right?

**Comment by cbielow** [comment]
> I think we moved to cython 3 right?

True: https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/requirements_bld.txt#L6

**Comment by timosachsenberg** [comment]
Done
